src/parser/statement.js

```diff
@@ -303,31 +304,39 @@ pp.parseForStatement = function (node) {
     return this.parseFor(node, init);
   }
 
+  let usedNowKeyword = false;
+  if (this.hasPlugin("lightscript") && (!forAwait)) {
+    // idx, elem, key, or val begins a LS-enhanced loop. The lookahead token must be a
+    // var name. If it isn't, someone is using one of the contextual keywords as a var name.
+    if (
+      (this.isContextual("idx") || this.isContextual("elem") || this.isContextual("key") || this.isContextual("val")) &&
+      (this.lookahead().type === tt.name)
```

Comment:
Right, sorry – we should have the same restriction for `for-of`, requiring `now` or `const`. In most cases, `for elem x in arr` should be used instead of `for x of arr` anyway, as it's faster... only when you actually have a generator on your hands should `for-of` be used, so requiring `const` should be a helpful nudge in the performant direction